staging/src/k8s.io/kms/internal/plugins/mock/kms.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mock-kmsv2-provider
+  namespace: kube-system
+  labels:
+    tier: control-plane
+    component: mock-kmsv2-provider
+spec:
+  hostNetwork: true
+  containers:
+    - name: mock-kmsv2-provider
+      image: localhost:5000/base64:e2e
+      imagePullPolicy: IfNotPresent
+      volumeMounts:
+        - name: sock
+          mountPath: /tmp
+  volumes:
+    - name: sock
+      hostPath:
+        path: /tmp
```

Comment:
```suggestion
      volumeMounts:
        - name: sock
          mountPath: /kms_socket
          readOnly: false  # writable to allow for socket use
  volumes:
    - name: sock
      emptyDir: {}
```

Reply:
@enj, Code is not able to create a socket file in emptyDir. Also, I tried several other locations before but none work other than /tmp. Let me know if I am missing anything here.

Reply:
I'll just live with the questionable use of the `tmp` dir xD